corehq/apps/app_manager/xpath.py

```diff
@@ -69,21 +69,27 @@ def dot_interpolate(string, replacement):
     return new
 
 
-def interpolate_xpath(string, case_xpath=None, fixture_xpath=None, module=None, form=None):
-    """
-    Replace xpath shortcuts with full value.
-    """
-    if case_xpath is None and any([
+def _validate_case_reference(string, module, form, interpolate_dots=True):
+    bad_references = [
         '#case' in string,
         '#parent' in string,
         '#host' in string,
-        # DOT_INTERPOLATE_PATTERN throws false positives, so if it flags the string,
-        # verify that the string's dots would actually get replaced
-        re.search(DOT_INTERPOLATE_PATTERN, string) and dot_interpolate(string, "") != string,
-    ]):
+    ]
+    if interpolate_dots:
+        bad_references.append(re.search(DOT_INTERPOLATE_PATTERN, string) and dot_interpolate(string, "") != string)
+    if any(bad_references):
         # At the moment this function is only used by module and form filters.
         # If that changes, amend the error message accordingly.
         raise CaseXPathValidationError(_(CASE_REFERENCE_VALIDATION_ERROR), module=module, form=form)
+
+
+def interpolate_xpath(string, case_xpath=None, fixture_xpath=None, module=None, form=None, interpolate_dots=True):
+    """
+    Replace xpath shortcuts with full value.
+    """
+    if case_xpath is None:
+        _validate_case_reference(string, module, form, interpolate_dots)
```

Comment:
nit: what about something like
```suggestion
        _ensure_no_case_references(string, module, form, interpolate_dots)
```

Reply:
👍 https://github.com/dimagi/commcare-hq/pull/33302/commits/ae4e0d25acee5fd4f72a4c9b6c3fa27c7599d4a2